### PR TITLE
pools: Add PalletIndex type

### PR DESCRIPTION
**Pull request by NunoAlexandre**
# Description

Currently, all of our native assets are described in `MultiLocation` as `X2(Parachain(_), GeneralKey(y))`. This works well for all assets with a static general key since we can _match_ on those static, const values. However, with Tranche tokens, as their general key is only obtained at runtime, we don't have a way of matching a MultiLocation representing a tranche apart from an exclusion-based approach.

Since we don't have tranche assets being used in production yet, hereby I suggest that we change the MultiLocation we use for tranche assets to be `X3(Parachain(x), PalletInstance(<pools_pallet_index>), GeneralKey(y))`, making it super easy to match on tranche tokens.

This comes particularly helpful in order to easily block transfers made through PolkadotXcm or XTokens that involve tranche tokens, which are disallowed to avoid transfers to non-whitelisted destination addresses. We now block all the extrinsics of those two pallets that allow for an arbitrary number of assets to be transferred, which we need to have enabled for USDC transfers where GLRM also needs to be sent alongside to cover for fee costs.

## Changes and Descriptions

1. We change the multilocation to which tranche tokens are represented as
2. We pass a `PalletIndex` type param to the pools pallet

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

^ I mean, in theory it's a breaking change but in practise I don't believe it's breaking anything.

# How Has This Been Tested?

We have adapted the affected tests and mocks in the pools pallet to cover this change.

# Checklist:

- [x] I have added Rust doc comments to structs, enums, traits and functions
- [x] I have made corresponding changes to the documentation
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I rebased on the latest `main` branch

### CODE REVIEW

**Comment by NunoAlexandre** [comment]
@mustermeiszer @offerijns can we get your reviews on this one please 🙏

File: pallets/loans/src/mock.rs [hunk 1]
```diff
@@ -152,6 +152,9 @@ impl orml_tokens::Config for MockRuntime {
 parameter_types! {
 	pub const PoolPalletId: frame_support::PalletId = cfg_types::ids::POOLS_PALLET_ID;
 
+	/// The index with which this pallet is instantiated in this runtime.
+	pub PoolPalletIndex: u8 = <Pools as PalletInfoAccess>::index() as u8;
```
Comment (mustermeiszer) [inline pallets/loans/src/mock.rs:156]: What size is `Index` that we need to cast it? Will this be safe? Can we have test in case this is overflowing? ^^

**Review by mustermeiszer — APPROVED** [review]
If you think this will be a good decision I trust you and it looks reasonable to me! 

The `GeneralKey` should then be unique across our Domain, right? At least in the context of the given `PalletInstance`, right?

**Comment by mustermeiszer** [comment]
No, I like the `X3`! In general XCM seems to be really bad ad addressing tokens currently. At least without using `GeneralIndex` and `GeneralKey`...

**Pull request merged**